editor/src/components/canvas/canvas-strategies/strategies/group-helpers.ts

```diff
@@ -21,30 +22,76 @@ import { importAlias } from '../../../../core/shared/project-file-types'
 import { assertNever } from '../../../../core/shared/utils'
 import { styleStringInArray } from '../../../../utils/common-constants'
 import { isCSSNumber } from '../../../inspector/common/css-utils'
+import { replaceNonDOMElementPathsWithTheirChildrenRecursive } from './fragment-like-helpers'
 
+// Returns true if the element should be treated as a group,
+// even if it's configuration (including its children) means that we cannot do any
+// group manipulation operations.
 export function treatElementAsGroupLike(
+  metadata: ElementInstanceMetadataMap,
+  path: ElementPath,
+): boolean {
+  return MetadataUtils.isGroupAgainstImports(MetadataUtils.findElementByElementPath(metadata, path))
+}
+
+// Determines if the element can be trued up as a group depending on how it has been configured.
+export function allowGroupTrueUp(
   metadata: ElementInstanceMetadataMap,
   pathTrees: ElementPathTrees,
+  allElementProps: AllElementProps,
   path: ElementPath,
 ): boolean {
-  const allChildrenAreAbsolute = MetadataUtils.getChildrenOrdered(metadata, pathTrees, path).every(
-    (child) => child.specialSizeMeasurements.position === 'absolute',
-  )
-  return (
-    allChildrenAreAbsolute &&
-    MetadataUtils.isGroupAgainstImports(MetadataUtils.findElementByElementPath(metadata, path))
-  )
+  const isGroupLike = treatElementAsGroupLike(metadata, path)
+  if (isGroupLike) {
+    const groupValidity = getGroupValidity(path, metadata, pathTrees, allElementProps)
+    switch (groupValidity) {
+      case 'valid':
+      case 'warning':
+        return true
+      case 'error':
+        return false
+      default:
+        assertNever(groupValidity)
+    }
+  } else {
+    return false
```

Comment:
ultra nitpick but this would be nicer as an early return above the happy path to the left